velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#define XXH_INLINE_ALL
+
+#include "velox/functions/prestosql/aggregates/PrestoHasher.h"
+#include "velox/external/xxhash.h"
+#include "velox/functions/lib/LambdaFunctionUtil.h"
+
+namespace facebook::velox::aggregate {
+
+namespace {
+
+template <typename T>
+FOLLY_ALWAYS_INLINE int64_t hashInteger(const T& value) {
+  return XXH64_round(0, value);
+}
+
+FOLLY_ALWAYS_INLINE int64_t
+hashStringView(const DecodedVector& vector, vector_size_t row) {
+  auto input = vector.valueAt<StringView>(row);
+  return XXH64(input.data(), input.size(), 0);
+}
+
+template <typename Callable>
+FOLLY_ALWAYS_INLINE void applyHashFunction(
+    const SelectivityVector& rows,
+    const DecodedVector& vector,
+    BufferPtr& hashes,
+    Callable func) {
+  VELOX_CHECK_GE(hashes->size(), rows.end())
+  auto rawHashes = hashes->asMutable<int64_t>();
+
+  rows.applyToSelected([&](auto row) {
+    if (vector.isNullAt(row)) {
```

Comment:
Thank you for consolidating the null checks in one place. This will make it easier to add an optimization to skip null checks if vector.mayHaveNulls() is false.